src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/DungeonNpcProfitOverlay.java

```diff
@@ -0,0 +1,176 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.miscfeatures;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+import io.github.moulberry.notenoughupdates.events.SlotClickEvent;
+import io.github.moulberry.notenoughupdates.mixins.AccessorGuiContainer;
+import io.github.moulberry.notenoughupdates.util.ItemResolutionQuery;
+import io.github.moulberry.notenoughupdates.util.ItemUtils;
+import io.github.moulberry.notenoughupdates.util.SBInfo;
+import io.github.moulberry.notenoughupdates.util.Utils;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.client.gui.inventory.GuiChest;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumChatFormatting;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.event.GuiOpenEvent;
+import net.minecraftforge.client.event.GuiScreenEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import org.lwjgl.opengl.GL11;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.github.moulberry.notenoughupdates.util.GuiTextures.dungeon_chest_worth;
+
+public class DungeonNpcProfitOverlay {
+
+	public static final ResourceLocation dungeonProfitResource =
+		new ResourceLocation("notenoughupdates:dungeon_chest_worth.png");
+
+	enum DungeonChest {
+		WOOD, GOLD, DIAMOND, EMERALD, OBSIDIAN, BEDROCK;
+	}
+
+	static class ChestProfit {
+		private DungeonChest dungeonChest;
+		private boolean profit = false;
+		private final List<String> items = new ArrayList<>();
+	}
+
+	private final Pattern pattern = Pattern.compile(".+ Catacombs - Floor .+");
+
+	private final List<ChestProfit> chestProfits = new ArrayList<>();
+	private ItemStack last;
+
+	@SubscribeEvent
+	public void onDrawBackground(GuiScreenEvent.BackgroundDrawnEvent event) {
+		if (!(event.gui instanceof GuiChest)) return;
+		String lastOpenChestName = SBInfo.getInstance().lastOpenChestName;
+		Matcher matcher = pattern.matcher(lastOpenChestName);
+		if (!matcher.matches()) return;
+
+		GuiChest guiChest = (GuiChest) event.gui;
+		if (guiChest.inventorySlots.getSlot(10).getStack() == last) {
+			render(guiChest);
+			return;
+		}
+		System.out.println("further");
```

Comment:
me when further

Reply:
yes it's not finished yet lulonaut will be *remaking* it